sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -4270,28 +4270,45 @@ export class BlockBlobClient extends BlobClient {
 
         const blockList: string[] = [];
         const blockIDPrefix = randomUUID();
-        let transferProgress: number = 0;
+        // Stores the amount of bytes progressed in each block
+        let transferProgressPerBlock: number[] = [];
```

Comment:
linter error `prefer-const`
```suggestion
        const transferProgressPerBlock: number[] = [];
```

Reply:
suggestion committed